### PR TITLE
Budget Rails App

### PR DESCRIPTION
**Pull request by cmolinan**
#### Design
- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

> NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.

Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the structure of an ERB diagram provide

### CODE REVIEW

**Comment by cmolinan** [comment]
Dear Code Reviewer, as a newcomer to Rails, I've tried my best to complete this project, including the extremely difficult deployment on render.com. I'm really exhausted. 
Thanks for your review!

File: README.md [hunk 1]
```diff
@@ -90,196 +67,133 @@ After you're finished please remove all the comments and instructions!
...
-> Describe between 1-3 key features of the application.
 
-- **[key_feature_1]**
-- **[key_feature_2]**
-- **[key_feature_3]**
+<br>
 
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
+## Video Presentation 
+[Video Demo Link](https://drive.google.com/drive/u/0/folders/1BUFwAkKcI3iUr-kRtgKTpWBJNlCVFNui)
```
Comment (OLIPLICHE) [inline README.md]: - Please provide an easier way to access the video. Kindly make the link available and easy to access to video

File: app/views/entities/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                <div class="item-upper">
+                  <div><%= t.name %></div>
+                  <div class='item-amount'>$<%= number_with_precision(t.amount, precision: 2) %></div>              
+                </div>
+                <div class="item-date">
+                  <%= t.created_at.strftime('%d %b %Y') %>
+                </div>
+              </div>
+            </div>
+
```
Comment (OLIPLICHE) [inline app/views/entities/index.html.erb:24]: Nice 👍

File: app/views/shared/_group.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+<div class='category-entitie'>
+  <div class='category-image'>
+      <div class='category-name'><%= group.name %></div> 
+      <div class='category-img'><img src='<%= image_path(group.icon) %>' alt='<%= group.name %>'/></div>
+  </div>
+  <div class='category-info'>
+    <div>Total: </div>
+    <div>$<%= number_with_precision(group.total_amount, precision: 2) %></div>
+    <% if false %>
+      <div>$<%= group.sum_total %> </div>
```
Comment (OLIPLICHE) [inline app/views/shared/_group.html.erb:11]: Nice 👍

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (OLIPLICHE) [inline spec/models/user_spec.rb:3]: Nice 👍

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @cmolinan , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- The App is working fine 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (cmolinan) [reply README.md]: @OLIPLICHE Thanks for your review and comments.  

I have uploaded the video to youtube, attach the link in the readme file.

File: app/views/entities/index.html.erb (hunk 2 above)
Comment (cmolinan) [reply app/views/entities/index.html.erb:24]: @OLIPLICHE Thanks !

File: app/views/shared/_group.html.erb (hunk 3 above)
Comment (cmolinan) [reply app/views/shared/_group.html.erb:11]: @OLIPLICHE Thanks !

File: spec/models/user_spec.rb (hunk 4 above)
Comment (cmolinan) [reply spec/models/user_spec.rb:3]: @OLIPLICHE Thanks !

File: README.md (hunk 1 above)
Comment (peterrobert) [reply README.md]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @cmolinan , 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cmolinan** [comment]
@peterrobert Thank you for your approval !

**Pull request merged**